src/unittest/test_clientactiveobjectmgr.cpp

```diff
@@ -58,85 +60,74 @@ class TestClientActiveObjectMgr : public TestBase
 
 	void runTests(IGameDef *gamedef);
 
-	void testFreeID();
-	void testRegisterObject();
-	void testRemoveObject();
 	void testGetActiveSelectableObjects();
 };
 
 static TestClientActiveObjectMgr g_test_instance;
 
 void TestClientActiveObjectMgr::runTests(IGameDef *gamedef)
 {
-	TEST(testFreeID);
-	TEST(testRegisterObject)
-	TEST(testRemoveObject)
 	TEST(testGetActiveSelectableObjects)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 
-void TestClientActiveObjectMgr::testFreeID()
```

Comment:
what is the plan if any converted tests need to access the gamedef?

Reply:
Good question. The plan is to move the `TestGameDef` to its own file and instantiate an instance of it for each Catch test that needs it. If that proves to be too slow, then I think mocking some parts of the gamedef might be appropriate. That's more work than a global gamedef, which almost looks like an amazing solution, but there's a big danger with that because the global instance might be modified by a test and not completely reset at the end. The tests would no longer be independent of each other and the order in which the tests are run could start to matter, which is a big mess. Looking at the code, it seems we could already have this problem.